packages/wrangler/src/ai/fetcher.ts

```diff
@@ -0,0 +1,24 @@
+import {getAccountId} from "../user";
+import {performApiFetch} from "../cfetch/internal";
+import { Request, Response } from "miniflare"
+
+export async function AIFetcher(request: Request) {
+	const accountId = await getAccountId();
+
+	const headers: Record<string, string> = {}
+	for (const [name, value] of request.headers) {
+		headers[name] = value
+	}
+
+	delete headers["host"]
+	delete headers["content-length"]
```

Comment:
This might be a little more ergonomic 🙂 
```suggestion
	const headers = new Headers(request.headers);
	headers.delete("host");
	headers.delete("content-length");
```

Reply:
Couldn't get it to work with `new Headers(request.headers);`  for some reason, had to revert de pr, at least for now

Reply:
Ah, you'll need to import it, like you're doing with Request/Response

Reply:
couldn't use`request.headers` directly in the fetch due to the same error, i had to convert it into an object using `Object.fromEntries(request.headers.entries()),`